### DISCUSSION TITLE
Value fluctuating

### DISCUSSION THREAD

**Discussion by Stipena** [jonas-merkle/AS5047P#23] [Q&A]
as5047p.readAngleRaw() value fluctuating somewhere around +-3 when no rotation is given. Is sensor that inaccurate or something else is happening? How to negotiate fluctuation to get stable readings?

**Comment by jonas-merkle** [comment]
I can confirm this behavior in general. In my test setup, I could sometimes observe it as well.
However, I don't know if these are actually measurement inaccuracies or if it's because the sensor responds so accurately with 14-bit resolution and such fluctuations might just be due to vibrations. I could also imagine an influence of the coils in the stepper motor if the stepper motor is actively holding position. Alternatively, it could be that the magent is not ideally aligned with the sensor.
But all this is just conjecture!

The function `readAngleRaw()` reads by default the measured value from the sensor, which has already passed an error correction in the sensor. To get the actual unprocessed raw value from the sensor the function call must be extended as follows: `readAngleRaw(false)` ([see](https://jonas-merkle.github.io/AS5047P/docs/master/html/class_a_s5047_p.html#a88982d7976889d4640cc81589c96473f)).
I don't know what the automatic angle correction in the sensor does exactly, you would have to read the data sheet. Generally there is additional information about the noise behavior of the sensor in the data sheet. I have just never looked at it more closely.

**Reply by Stipena** [reply]
Vibrations and influence of coils isn't problem since I made separate case to test sensor without motor. Tried "readAngleRaw(false)", nothing changed, exact same values. Did I get this right, from quote below, that disabling DAEC increases standstill readings?

> It is also possible to disable the Dynamic Angle Error
Compensation with the setting DAECDIS. Disabling the
Dynamic Angle Error Compensation gives a noise benefit of
0.016 degree rms.This setting can be advantageous for low
speed (under 100rpm) respectively static positioning
applications.

**Reply by jonas-merkle** [reply]
Yes, according to the quote, a measurement at standstill is more accurate when DAEC is disabled, that is, when `readAngleRaw(false)` is used. However, I'm not surprised that you don't see any improvement in the values. If you normally have a variation of +-3 it means that the value varies by +-3*360/2^14 = +-0.0659... degrees and disabling DAEC results in an improvement of only 0.016 degrees.

**Reply by Stipena** [reply]
I will try to ask AMS about this fluctuating behavior. Thanks for help!

**Reply by Stipena** [reply]
AMS reply about fluctuation:

> A peak to peak noise from +/-3 counts (in total 6 counts) is normal and nothing to worry about. 
> Since AS5047P has got 14 Bit output this only represents 0.0659° degree peak to peak noise which is totally in our spec. limit.
> 
> If you need a sensor who does not have this fluctuation while the magnet is on a fixed position you can refer to our AS5600 or AS5600L. This sensor is using I2C interface and has got a 12 Bit angle output but has a decreased noise due to internal filtering.
